filebeat/input/v2/statestore/store.go

```diff
@@ -0,0 +1,132 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package statestore
+
+import (
+	"sync"
+
+	"github.com/elastic/beats/libbeat/logp"
+	"github.com/elastic/beats/libbeat/registry"
+)
+
+// Store provides some coordinates access to a registry.Store.
+// All update and read operations require users to acquire an resource first.
+// A Resource must be locked before it can be modified. This ensures that at most
+// one go-routine has access to a resource. Lock/TryLock/Unlock can be used to
+// coordinate resource access even between independent components.
+type Store struct {
+	active bool
```

Comment:
Should active be an atomic value? Close/Desactive and Creating of key can both access or change the state of this value?